zulipterminal/model.py

```diff
@@ -301,10 +303,34 @@ def set_narrow(
                 break
         else:
             raise RuntimeError("Model.set_narrow parameters used incorrectly.")
+        
+        try:
+            if self.stream_id:
+                self.active_button = self.controller.view.stream_id_to_button[
+                    self.stream_id
+                ]
```

Comment:
If the narrow changes to a non-stream one, this seems like we're setting the button to the old stream one?

Reply:
Perhaps the active_button should rather be last_active_stream?

Reply:
Perhaps the larger point here is really that we're changing the button, before we know if the narrow changed.